### PR TITLE
migration: disabling vault sa based auth for storage class encryption

### PR DESCRIPTION
**Pull request by debjyoti-pandit**
This PR contains the migrations of this PR https://github.com/openshift/console/pull/10998

Description: https://github.com/openshift/console/pull/10998/files#issue-1122752788

### CODE REVIEW

File: packages/odf/components/kms-config/utils.tsx [hunk 1]
```diff
@@ -252,55 +271,74 @@ const createCsiHpcsResources = (
     IBM_KP_BASE_URL: kms.baseUrl.value,
     IBM_KP_TOKEN_URL: kms.tokenUrl,
   };
-  const csiConfigObj: ConfigMapKind = generateCsiKmsConfigMap(kms.name.value, csiConfigData, false);
+  const csiConfigObj: ConfigMapKind = generateCsiKmsConfigMap(
+    kms.name.value,
+    csiConfigData
+  );
```
Comment (SanjalKatiyar) [inline packages/odf/components/kms-config/utils.tsx:277]:
```suggestion
  const csiConfigObj: ConfigMapKind = generateCsiKmsConfigMap(
    kms.name.value,
    csiConfigData,
    false
  );
```

File: packages/odf/components/kms-config/utils.tsx [hunk 2]
```diff
@@ -252,55 +271,74 @@ const createCsiHpcsResources = (
...
+
   if (update) {
-    const cmPatch = [generateConfigMapPatch(kms.name.value, csiConfigData, false)];
-    csiKmsResources.push(k8sPatch({model: ConfigMapModel, resource: csiConfigObj, data: cmPatch}));
+    const cmPatch = [generateConfigMapPatch(kms.name.value, csiConfigData)];
```
Comment (SanjalKatiyar) [inline packages/odf/components/kms-config/utils.tsx:280]:
```suggestion
    const cmPatch = [generateConfigMapPatch(kms.name.value, csiConfigData, false)];
```

File: packages/odf/components/kms-config/utils.tsx [hunk 3]
```diff
@@ -134,9 +135,7 @@ const generateHpcsSecret = (kms: HpcsConfig) => ({
   apiVersion: SecretModel.apiVersion,
   kind: SecretModel.kind,
   metadata: {
-    name: `ibm-kp-kms-${Math.random()
-      .toString(36)
-      .substring(7)}`,
+    name: `ibm-kms-key-${Math.random().toString(36).substring(7)}`,
```
[the remark this answers is not in the document]
Comment (debjyoti-pandit) [reply packages/odf/components/kms-config/utils.tsx]: taken care

File: packages/odf/components/kms-config/utils.tsx (hunk 1 above)
Comment (debjyoti-pandit) [reply packages/odf/components/kms-config/utils.tsx]: taken care

**Pull request merged**